File: csharp/matrix/Matrix.cs
using System.Linq;

public class Matrix
{
    private readonly int[][] _matrix;

    public Matrix(string input) => _matrix = input.Split('\n').Select(r => r.Split().Select(int.Parse).ToArray()).ToArray();

    public int[] Row(int row) => _matrix[row - 1];

    public int[] Column(int col) => _matrix.Select(r => r[col - 1]).ToArray();
}

File: solutions/kotlin/matrix/1/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {

    fun column(colNr: Int): List<Int> = matrixAsString.split('\n').map { it.split(Regex(" +")).elementAt(colNr - 1) }.map { it.toInt() }

    fun row(rowNr: Int): List<Int> = matrixAsString.split('\n').elementAt(rowNr - 1).split(Regex(" +")).map { it.toInt() }
}
